### PR TITLE
Always emit WireField annotations for Kotlin properties

### LINKED ISSUE #780: Add WireField annotations for non-interop mode
`WireField` annotations are only emitted in `java_interop` mode, do we need them in Kotlin-only mode? Should we put them behind a custom flag? Do we need them at all?

### PR DESCRIPTION
**Pull request by Egorand**
Fixes #780

### CODE REVIEW

File: wire-tests/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/person/Person.kt [hunk 1]
```diff
@@ -21,10 +22,11 @@ import kotlin.jvm.JvmStatic
 import okio.ByteString
 
 data class Person(
-    val name: String,
-    val id: Int,
-    val email: String? = null,
-    val phone: List<PhoneNumber> = emptyList(),
+    @WireField(tag = 1, adapter = "com.squareup.wire.ProtoAdapter.STRING") val name: String,
+    @WireField(tag = 2, adapter = "com.squareup.wire.ProtoAdapter.INT32") val id: Int,
+    @WireField(tag = 3, adapter = "com.squareup.wire.ProtoAdapter.STRING") val email: String? =
```
Comment (jrodbx) [inline wire-tests/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/person/Person.kt]: nit: `val email...` on new line

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt [hunk 2]
```diff
@@ -377,12 +377,10 @@ class KotlinGenerator private constructor(
...
-      if (javaInterOp) {
-        parameterSpec.addAnnotation(AnnotationSpec.builder(WireField::class)
-            .addMember("tag = %L", field.tag())
-            .addMember("adapter = %S", getAdapterName(field))
-            .build())
-      }
+      parameterSpec.addAnnotation(AnnotationSpec.builder(WireField::class)
+          .addMember("tag = %L", field.tag())
+          .addMember("adapter = %S", getAdapterName(field))
+          .build())
```
Comment (jrodbx) [inline wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: Should the annotation be limited to the field element via [use-site targets](https://kotlinlang.org/docs/reference/annotations.html#annotation-use-site-targets)?

File: wire-tests/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/person/Person.kt (hunk 1 above)
Comment (Egorand) [reply wire-tests/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/person/Person.kt]: Yeah, that's KotlinPoet

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt (hunk 2 above)
Comment (Egorand) [reply wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: Yeah, guess that makes sense

**Pull request merged**